### PR TITLE
Added checks for ID3D12InfoQueue1

### PR DESCRIPTION
**Pull request by NipunRamani99**

### CODE REVIEW

File: includes/D3D/D3DDebugLogger.hpp [hunk 1]
```diff
@@ -13,14 +13,12 @@ class D3DDebugLogger {
...
 	void UnregisterCallBack() const noexcept;
 
 	static void MessageCallback(
 		D3D12_MESSAGE_CATEGORY category, D3D12_MESSAGE_SEVERITY severity,
 		D3D12_MESSAGE_ID id, LPCSTR pDescription, void* pContext
 	);
-#endif
 private:
 #ifdef __ID3D12InfoQueue1_INTERFACE_DEFINED__
 	DWORD m_callBackCookie;
```
Comment (razerx100) [inline includes/D3D/D3DDebugLogger.hpp:24]: No need to put this one under the macro. Then you didn't have to touch its initialisation in the ctor.

File: src/D3D/D3DDebugLogger.cpp [hunk 2]
```diff
@@ -27,6 +27,7 @@ static std::array<const char*, 5u> messageSeverity{
 D3DDebugLogger::D3DDebugLogger(const Args& arguments) {
 
 #ifdef __ID3D12InfoQueue1_INTERFACE_DEFINED__
+	m_callBackCookie = 0;
```
Comment (razerx100) [inline src/D3D/D3DDebugLogger.cpp:30]: I am talking about this. This is copy assign, not constructing. Kind of a nitpick though.

File: src/D3D/D3DDebugLogger.cpp [hunk 3]
```diff
@@ -46,16 +40,22 @@ D3DDebugLogger::D3DDebugLogger(const Args& arguments) : m_callBackCookie{ 0u } {
 	else {
 		std::ofstream{ "ErrorLog.txt", std::ios_base::app | std::ios_base::out } << std::format("Something went wrong while initializing debugInfoQueue. HRESULT:0x{0:X}", static_cast<unsigned>(hr)).c_str() << std::endl;
 	}
+#else
+	OutputDebugStringA("Failed to initalize D3D12DebugLogger: ID3D12InfoQueue1 interface is not defined. Upgrade to Windows 11 or use DX12 Agility SDK from NuGet.\n");
+	std::ofstream{ "ErrorLog.txt", std::ios_base::app | std::ios_base::out } << "Failed to initalize D3D12DebugLogger: ID3D12InfoQueue1 interface is not defined. Upgrade to Windows 11 or use DX12 Agility SDK from NuGet." << std::endl;
```
Comment (razerx100) [inline src/D3D/D3DDebugLogger.cpp:46]: Duplicate string. Should have put it into a variable/constant and used that instead.

File: includes/D3D/D3DDebugLogger.hpp (hunk 1 above)
Comment (NipunRamani99) [reply includes/D3D/D3DDebugLogger.hpp:25]: done

**Pull request merged**